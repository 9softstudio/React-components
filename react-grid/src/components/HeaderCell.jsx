import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class HeaderCell extends Component {
    constructor(props) {
        super(props);

    }

    static propTypes = {
        sortOrder: PropTypes.string,
        sortBy: PropTypes.string,
        onSort: PropTypes.func
    }
 
    static defaultProps = {
        sortOrder: null,
        sortBy: null,
        onSort: null
    }

    onClick = () => {
        if(!this.props.sortBy){
            return;
        }
        const { onSort } = this.props;
        const sortOrder = this.props.sortOrder;

        const newSortOrder = !sortOrder || sortOrder === 'desc' ? 'asc' : 'desc';
        onSort && onSort(this.props.sortBy, newSortOrder);
    }

    render() {
        const { colWidth, sortBy, sortOrder, onSort, children, ...rest } = this.props;
    
        let sortableHeader = null;
        if(sortBy) {
            const upArrow = <div style={{display:'block', color: `${sortOrder === 'asc' ? '#111':'#777'}`}} className={'asc'}></div>
            const downArrow = <div style={{display:'block', marginTop:'2px', color: `${sortOrder === 'desc' ? '#111':'#777'}`}} className={'desc'}></div>
            sortableHeader = (<th {...rest} className={'sortable sorting'} style={{position:'relative', backgroundClip: 'padding-box'}} onClick={this.onClick}>                   
                <div style={{width:'90%'}}>{children}</div>
                <span style={{float: 'right', position:'absolute', top:'40%', right:'2px'}}>
                    {upArrow}
                    {downArrow}
                </span>
                
            </th>)
        }

        return sortableHeader || 
            (<th {...rest}>{children}</th>)
    }
}