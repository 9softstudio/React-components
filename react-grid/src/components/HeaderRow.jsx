import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class HeaderRow extends Component {
    constructor(props) {
        super(props)
    }

    static propTypes = {
        onSort: PropTypes.func
    }
 
    static defaultProps = {
        onSort: null
    }

    _renderChildren() {
        const { onSort, children, sortBy, sortOrder } = this.props;
        
        // register onSort event
        return React.Children.map(children, child => {
            return child && React.cloneElement(child, { onSort: onSort, sortOrder: sortBy === child.props.sortBy ? sortOrder : null});
        });
    }

    render() {
        const { onSort, sortBy, sortOrder, ...rest } = this.props;
        return (
            <tr {...rest}>{this._renderChildren()}</tr>
        );
    }
}