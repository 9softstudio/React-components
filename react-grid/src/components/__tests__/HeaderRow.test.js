import React from 'react'
import HeaderRow from '../HeaderRow'
import HeaderCell from '../HeaderCell'
import ShallowRenderer from 'react-test-renderer/shallow'

test('HeaderRow component render correctly', () => {
    const renderer = new ShallowRenderer();

    const header = renderer.render(
        <HeaderRow sortBy='Cell1' sortOrder='asc'>
            <HeaderCell sortBy='Cell1'>Cell 1</HeaderCell>
            <HeaderCell>Cell 2</HeaderCell>
        </HeaderRow>
    );

    expect(header).toMatchSnapshot();
})