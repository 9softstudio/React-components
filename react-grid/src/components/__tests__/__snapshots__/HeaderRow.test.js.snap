// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRow component render correctly 1`] = `
<tr>
  <HeaderCell
    onSort={null}
    sortBy="Cell1"
    sortOrder="asc"
  >
    Cell 1
  </HeaderCell>
  <HeaderCell
    onSort={null}
    sortBy={null}
    sortOrder={null}
  >
    Cell 2
  </HeaderCell>
</tr>
`;
